src/librustc_lint/nonstandard_taste.rs

```diff
@@ -0,0 +1,52 @@
+use crate::{LateContext, LateLintPass, LintContext};
+use rustc_hir as hir;
+
+declare_lint! {
+    pub PINEAPPLE_ON_PIZZA,
+    Forbid,
+   "pineapple doesn't go on pizza"
+}
+
+declare_lint_pass!(PineappleOnPizza => [PINEAPPLE_ON_PIZZA]);
+
+impl<'a, 'tcx> LateLintPass<'a, 'tcx> for PineappleOnPizza {
+    fn check_ty(&mut self, cx: &LateContext<'_, '_>, ty: &hir::Ty<'_>) {
+        if let hir::TyKind::Path(hir::QPath::Resolved(_, path)) = &ty.kind {
+            for pizza_segment in path.segments {
+                if let Some(args) = pizza_segment.args {
+                    if pizza_segment.ident.name.as_str().to_lowercase() != "pizza" {
+                        continue;
+                    }
+                    for arg in args.args {
+                        if let hir::GenericArg::Type(hir::Ty {
```

Comment:
This should be extended to support const generics:

```rust
Pizza<pineapple()>
```

should not be allowed, since there is no *value* to having pineapple on pizza.